From 17573b3d1959ccb53685635937f883e41f21546e Mon Sep 17 00:00:00 2001
From: Carsten Wolff <carsten@wolffcarsten.de>
Date: Sun, 8 Apr 2018 21:39:10 +0200
Subject: [PATCH] iputils: install ping, ping6, traceroute6 with setuid root

these utilities need to run with uid 0 to be useful. Thus,
install them setuid root like other distros do, too.

Signed-off-by: Carsten Wolff <carsten@wolffcarsten.de>
---
 package/network/utils/iputils/Makefile | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/package/network/utils/iputils/Makefile b/package/network/utils/iputils/Makefile
index 344af07afd4..e865b6e3da0 100644
--- a/package/network/utils/iputils/Makefile
+++ b/package/network/utils/iputils/Makefile
@@ -144,12 +144,12 @@ endef
 
 define Package/iputils-ping/install
 	$(INSTALL_DIR) $(1)/usr/bin
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/ping $(1)/usr/bin/
+	install -m4755 $(PKG_BUILD_DIR)/ping $(1)/usr/bin/
 endef
 
 define Package/iputils-ping6/install
 	$(INSTALL_DIR) $(1)/usr/bin
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/ping6 $(1)/usr/bin/
+	install -m4755 $(PKG_BUILD_DIR)/ping6 $(1)/usr/bin/
 endef
 
 define Package/iputils-tftpd/install
@@ -169,7 +169,7 @@ endef
 
 define Package/iputils-traceroute6/install
 	$(INSTALL_DIR) $(1)/usr/bin
-	$(INSTALL_BIN) $(PKG_BUILD_DIR)/traceroute6 $(1)/usr/bin/
+	install -m4755 $(PKG_BUILD_DIR)/traceroute6 $(1)/usr/bin/
 endef
 
 $(eval $(call BuildPackage,iputils-arping))
